cmd/ube-apiserver/app/server.go

```diff
@@ -0,0 +1,589 @@
+			}
+			name, err := os.Hostname()
+			if err != nil {
+				glog.Fatalf("Failed to get hostname: %v", err)
+			}
+			addrs, err := instances.NodeAddresses(name)
+			if err != nil {
+				glog.Warningf("Unable to obtain external host address from cloud provider: %v", err)
+			} else {
+				for _, addr := range addrs {
+					if addr.Type == api.NodeExternalIP {
+						s.ExternalHost = addr.Address
+					}
+				}
+			}
+		}
+	}
+
+	config := &master.Config{
+		Config: &genericapiserver.Config{
+			StorageDestinations:       storageDestinations,
+			StorageVersions:           storageVersions,
+			ServiceClusterIPRange:     &n,
+			EnableLogsSupport:         s.EnableLogsSupport,
+			EnableUISupport:           true,
+			EnableSwaggerSupport:      true,
+			EnableProfiling:           s.EnableProfiling,
+			EnableWatchCache:          s.EnableWatchCache,
+			EnableIndex:               true,
+			APIPrefix:                 s.APIPrefix,
+			APIGroupPrefix:            s.APIGroupPrefix,
+			CorsAllowedOriginList:     s.CorsAllowedOriginList,
+			ReadWritePort:             s.SecurePort,
+			PublicAddress:             s.AdvertiseAddress,
+			Authenticator:             authenticator,
+			SupportsBasicAuth:         len(s.BasicAuthFile) > 0,
+			Authorizer:                authorizer,
+			AdmissionControl:          admissionController,
+			APIGroupVersionOverrides:  apiGroupVersionOverrides,
+			MasterServiceNamespace:    s.MasterServiceNamespace,
+			MasterCount:               s.MasterCount,
+			ExternalHost:              s.ExternalHost,
+			MinRequestTimeout:         s.MinRequestTimeout,
+			ProxyDialer:               proxyDialerFn,
+			ProxyTLSClientConfig:      proxyTLSClientConfig,
+			ServiceNodePortRange:      s.ServiceNodePortRange,
+			KubernetesServiceNodePort: s.KubernetesServiceNodePort,
+			Serializer:                api.Codecs,
+		},
+		EnableCoreControllers:   true,
+		DeleteCollectionWorkers: s.DeleteCollectionWorkers,
+		EventTTL:                s.EventTTL,
+		KubeletClient:           kubeletClient,
+
+		Tunneler: tunneler,
+	}
+
+	if s.EnableWatchCache {
+		cachesize.SetWatchCacheSizes(s.WatchCacheSizes)
+	}
+
+	m, err := master.New(config)
+	if err != nil {
+		return err
+	}
+
+	addControlplaneAPIGroup(storageDestinations, s)
+	installClusterAPI(m, storageDestinations)
+
+	m.Run(s.ServerRunOptions)
+	return nil
+}
+
+func getRuntimeConfigValue(s *options.APIServer, apiKey string, defaultValue bool) bool {
+	flagValue, ok := s.RuntimeConfig[apiKey]
+	if ok {
+		if flagValue == "" {
+			return true
+		}
+		boolValue, err := strconv.ParseBool(flagValue)
+		if err != nil {
+			glog.Fatalf("Invalid value of %s: %s, err: %v", apiKey, flagValue, err)
+		}
+		return boolValue
+	}
+	return defaultValue
+}
+
+// Parses the given runtime-config and formats it into map[string]ApiGroupVersionOverride
+func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGroupVersionOverride, error) {
```

Comment:
This function as is is useless for ube server.
We can either add support to this for controlplane group.

Or better, I think we can just skip the runtime-config flag for ubernetes. We dont need it.